wasi/wasi.go

```diff
@@ -15,6 +15,75 @@ import (
 	"github.com/tetratelabs/wazero/wasm"
 )
 
+// Api is a documentation interface for WASI exported functions in version "wasi_snapshot_preview1"
+//
+// Note: In WebAssembly 1.0 (MVP), there may be up to one Memory per store, which means the precise memory is always
+// wasm.Store Memories index zero: `store.Memories[0].Buffer`
+//
+// See https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md
+// See https://www.w3.org/TR/wasm-core-1/#memory-instances%E2%91%A0.
+type Api interface {
```

Comment:
another pov: this is especially great as this provides a simple way for users to provide there own WAIS impls. I love this.

Reply:
sorry about my late comment, I think you just here mentioned the intent is to allow folks direct access. works for me! only thing we need to add to the interface is `Register(store *wasm.Store) (err error)`

Reply:
We have to think more about returning this to end users, as each function needs a `HostFunctionCallContext`. Probably we should turn that into an interface before exposing it as a return val or something. See #204